docs/library/machine.CAN.rst

```diff
@@ -0,0 +1,476 @@
+   that refers to a ``bytearray`` or similar object that has enough capacity to hold
+   any received CAN message (8 bytes for CAN Classic, 64 bytes for CAN FD). The
+   provided list will be returned as a successful result, and avoids memory
+   allocation inside the function.
+
+   If no messages have been received by the CAN controller, this function
+   returns ``None``.
+
+   Otherwise, it returns a list with 4 elements:
+
+   - Index 0 is the CAN ID of the received message, as an integer.
+   - Index 1 is a memoryview that provides access to the received message data.
+
+     - If *arg* is not provided then this is a ``memoryview`` holding the bytes
+       which were received. This ``memoryview`` is backed by a newly allocated
+       ``bytearray`` large enough to hold any received CAN message. This allows
+       the result to be safely reused as a future *arg*, to save memory allocations.
+     - If *arg* is provided then the provided ``memoryview`` will be resized to
+       hold exactly the bytes which were received. The caller is responsible for
+       making sure the backing object for the ``memoryview`` can hold a CAN
+       message of any length.
+
+   - Index 2 is an integer flags value bitwise ORed from
+     :class:`CAN.MessageFlags`. It provides metadata about the received message.
+   - Index 3 is an integer flags value bitwise ORed from
+     :class:`CAN.RecvErrors`. Any non-zero value indicates potential issues with
+     the receive functionality. These flags are reset
+     returns.
+
+.. method:: CAN.irq(handler=None, trigger=0, hard=False)
+
+   Sets an interrupt ``callback`` function to be called when one or more of the
+   events flagged in ``trigger`` has occurred.
+
+     - *handler* is a function to be called when the interrupt event
+       triggers.  The handler must take exactly one argument which is the
+       ``CAN`` instance.
+
+     - *trigger* configures the event(s) which can generate an interrupt.
+       Possible values are a mask of one or more of the following:
+
+       - ``CAN.IRQ_RX_DONE`` event occurs after the CAN controller has received
+         at least one message into its RX FIFO (meaning that :func:`CAN.recv()`
+         will return successfully).
+       - ``CAN.IRQ_TX_DONE`` event occurs after the CAN controller has either
+         successfully sent a message onto the CAN bus or failed to send a
+         message. This trigger has additional requirements for the handler, see
+         `machine_can_irq_flags` for details.
+       - ``CAN.IRQ_STATE`` interrupt if the CAN controller changes error states.
+
+     - *hard* if True, a hard interrupt is used. This reduces the delay between
+       the CAN controller event and the handler being called. Hard interrupt
+       handlers may not allocate memory; see :ref:`isr_rules`.
+
+   Returns an irq object. If called with no arguments then a previously-configured
+   irq object is returned.
+
+.. method:: CAN.cancel_send(index)
```

Comment:
Question, why support this at all? Generally, if you send a message on CAN you just expect it to be delivered...